docs: ✏️ add FioriAR 2.0 migration guide

MarcoEidinger opened the pull request:
Hi Patrick,

I wrote a migration guide in the wiki (https://github.com/SAP/cloud-sdk-ios-fioriarkit/wiki/FioriAR-2.0-Migration-Guide) and linked it in the README. Can you check the migration guide please? 

I tried to focus on the most important breaking changes. We have more, e.g

- ViewBuilder for `scanView` expects an enum
- `AnnotationLoadingStrategy` load now returns a tuple
- `ARManagement` Protocol Removed
- Renamed `descriptionText` to `subtitle`
- Changed CardItemModel` Data Types

but I feel not everything has to be covered in the example as the majority of developers probably won't be affected. But we maybe should list them in the migration guide. I am eager to hear your opinion.

Kind regards,
Marco

pgobriensap reviewed (approved):
It looks good to me! Concise yet clear description/comparison of what has been changed.

pgobriensap commented:
On a second thought. I agree that not everything needs to be mentioned, but a few thoughts.

1. ViewBuilder for `scanView` expects an enum
    * Perhaps a mention as its likely ViewBuilders will be used as we have seen evidence that Kevin used them in his demo. This is on the line and could go either way.
2. `AnnotationLoadingStrategy` load now returns a tuple
    * A large change, but I don't think it needs a mention as a custom strategy is highly unlikely
3. `ARManagement` Protocol Removed
    * Same as above, only visible through a custom strategy I believe
4. Renamed `descriptionText` to `subtitle`
    * Very subtle and straightforward, no mention needed
5. Changed `CardItemModel` Data Types
    * This will impact core usage and probably would feel strange if it's not mentioned, even if its straightforward

MarcoEidinger commented:
Thanks @pgobriensap for your feedback.  Based on your feedback I decided to mentioned all changes to `CardItemModel` protocol in the migration guide.

1. ViewBuilder for `scanView` expects an enum
    * ☑️ not mentioned. P.S.: in Kevin's [demo](https://youtu.be/kOFuwDSXBZg?t=1031) he is using `cardLabel` ViewBuilder.
2. `AnnotationLoadingStrategy` load now returns a tuple
    * ☑️ not mentioned
3. `ARManagement` Protocol Removed
    * ☑️ not mentioned
4. Renamed `descriptionText` to `subtitle`
    *  ✅ mentioned
5. Changed `CardItemModel` Data Types
    * ✅ mentioned
